hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -158,15 +161,21 @@ int currentPipelineCount(DatanodeDetails datanodeDetails, int nodesRequired) {
         ScmConfigKeys.OZONE_SCM_CONTAINER_SIZE_DEFAULT,
         StorageUnit.BYTES);
 
+    long metaSizeRequired = (long) conf.getStorageSize(
+        OZONE_DATANODE_RATIS_VOLUME_FREE_SPACE_MIN,
+        OZONE_DATANODE_RATIS_VOLUME_FREE_SPACE_MIN_DEFAULT,
+        StorageUnit.BYTES);
+
     // filter nodes that don't even have space for one container
     List<DatanodeDetails> canHoldList = healthyNodes.stream().filter(d ->
         hasEnoughSpace(d, sizeRequired)).collect(Collectors.toList());
 
     if (canHoldList.size() < nodesRequired) {
       msg = String.format("Pipeline creation failed due to no sufficient" +
-          " healthy datanodes with enough space for even a single container." +
-          " Required %d. Found %d. Container size %d.",
-          nodesRequired, canHoldList.size(), sizeRequired);
+          " healthy datanodes with enough space for container data or " +
```

Comment:
container data or metadata -> container data and metadata

Reply:
OK.